File: app-presentation/src/main/java/com/zxjdev/smile/presentation/authorization/login/LoginContract.java
package com.zxjdev.smile.presentation.authorization.login;

import com.zxjdev.smile.presentation.common.base.BasePresenter;
import com.zxjdev.smile.presentation.common.base.BaseView;

public interface LoginContract {

  interface View extends BaseView {

    void navigateToMain();

    void initUserComponent();
  }

  interface Presenter extends BasePresenter<View> {

    void handleLogin(String username, String password);
  }
}

File: app-presentation/src/main/java/com/zxjdev/smile/presentation/authorization/login/LoginContract.kt
package com.zxjdev.smile.presentation.authorization.login

import com.zxjdev.smile.presentation.common.base.BasePresenter
import com.zxjdev.smile.presentation.common.base.BaseView

interface LoginContract {

    interface View : BaseView {

        fun navigateToMain()

        fun initUserComponent()
    }

    interface Presenter : BasePresenter<View> {

        fun handleLogin(username: String, password: String)
    }
}
